index.html

```diff
@@ -559,53 +442,54 @@ <h3>
-          steps:
-        </p>
-        <ol class="algorithm" data-link-for="WakeLock">
-          <li>Set |lock:WakeLock|.{{[[requestPromise]]}} to `undefined`.
-          </li>
-          <li>Abort the associated <a>request()</a> algorithm.
-          </li>
-        </ol>
         <p data-tests="wakelock-api.https.html" data-link-for="WakeLock">
-          The <a>request()</a> method, when invoked, MUST run the following
-          steps:
+          The <a data-lt="request">request(|type:WakeLockType|,
+          |options:WakeLockRequestOptions|)</a> method, when invoked,
+          MUST run the following steps:
         </p>
         <ol class="algorithm">
-          <li>Let |options:WakeLockRequestOptions| be the first argument.
+          <li>Let |lockPromise:Promise| be <a data-cite=
+          "promises-guide#a-new-promise">a new promise</a>.
+          </li>
+          <li>Let |document:Document| be the <a>responsible document</a> of the
+          <a>current settings object</a>.
+          </li>
+          <li>If the <a>current global object</a> is the
+          {{DedicatedWorkerGlobalScope}} object and its <a>owner set</a>
+          <a data-xref-for="list">is empty</a>, reject |lockPromise| with a
+          "{{NotAllowedError}}" {{DOMException}} and abort these steps.
           </li>
-          <li>Let |lock:WakeLock| be the <a>context object</a>.
+          <li>If the <a>current global object</a> is the {{Window}} object and
+          the |document|'s [=Document/browsing context=] is `null`, reject
+          |lockPromise| with a "{{NotAllowedError}}" {{DOMException}} and
+          abort these steps.
           </li>
-          <li>Let |promise:Promise| be <a data-cite=
-            "promises-guide#a-new-promise">a new promise</a>.
+          <li data-tests="wakelock-disabled-by-feature-policy.https.sub.html">
+          If |document| is not <a>allowed to use</a> the <a>policy-controlled
+          feature</a> named "`wake-lock`", reject |lockPromise| with a
+          "{{NotAllowedError}}" {{DOMException}} and abort these steps.
           </li>
-          <li>If |lock|.{{[[requestPromise]]}} is not `undefined`,
-            reject |promise| with a "{{InvalidStateError}}" {{DOMException}},
-            and run <a>abort the request steps</a>.
+          <li>If the <a>user agent</a> <a data-lt="deny wake lock">denies the
+          wake lock</a> of this |type| for |document|, reject |lockPromise|
+          with a "{{NotAllowedError}}" {{DOMException}} and abort these steps.
           </li>
           <li>If |options|.|signal:AbortSignal|’s <a>aborted flag</a> is set, then
-          reject |promise| with an "{{AbortError}}" {{DOMException}},
-          and run <a>abort the request steps</a>.
+          reject |lockPromise| with an "{{AbortError}}" {{DOMException}},
+          and abort these steps.
           </li>
           <li>If |options|.|signal| is not `null`, then [=AbortSignal/add=] to |options|.|signal|:
```

Comment:
This is not strictly related to this PR, but I find the [example in the DOM spec](https://dom.spec.whatwg.org/#aborting-ongoing-activities-spec-example) easier to follow (i.e. rather than having two separate checks for `options.signal` have one with two branches).

Reply:
OK, let's fix that later

Reply:
Fixed!